rust/lance/src/dataset.rs

```diff
@@ -573,7 +654,8 @@ impl Dataset {
             manifest_location,
             commit_handler,
             session,
-            tags,
+            tags: refs.tags(),
```

Comment:
should we just store a `refs` in dataset instead of separated tags and branches?

Reply:
Ok, This is caught in a dilemma​. The original tag interface is like dataset.tag.create...list...delete. If we use refs here I think we shall use dataset.tags(). But somehow it brings backward compatibility.

The second solution is to put branches and tags together into refs and keep original tag interfaces and add branch operations.

But, the original tag operations is like create, list, delete, fetch. This will be strange if we have a create_branch() to create branch and a create() to create tag.

That's why it looks like this now

Reply:
> But somehow it brings backward compatibility.
 
what do you mean by that?

Reply:
I think I might misunderstand the meaning of backward compatibility.

What I want to say is the interface compatibility. Because we use dataset.tags for tag API, if we change that, the upper modules(java, python at least) or services like lancedb may need to adapt it too. What do you think of this?

Reply:
I think that's fine. We make breaking changes all the time, this is already a breaking change anyway